db/migrate/20190806102256_add_applications_open_from_to_courses.rb

```diff
@@ -0,0 +1,16 @@
+class AddApplicationsOpenFromToCourses < ActiveRecord::Migration[5.2]
+  def change
+    add_column :course, :applications_open_from, :date
+
+    say_with_time 'Setting applications_open_from to one held within sites' do
+      Course.all.each do |course|
+        if course.recruitment_cycle.current?
+          applications_open_from = course.site_statuses.order("applications_accepted_from ASC").first.applications_accepted_from
+          course.update(applications_open_from: applications_open_from)
+        else
+          course.update(applications_open_from: course.recruitment_cycle.application_start_date)
+        end
+      end
```

Comment:
I forget what Rails does when you try to roll this back, does it work (i.e. just drop the column and not complain)?

Reply:
Yep - it'll just drop it silently, this doesn't change any existing data so we don't need to worry about creating a more in-depth up/down migration

Reply:
Also, as you pointed out yourself, we should probably use `update!` here.